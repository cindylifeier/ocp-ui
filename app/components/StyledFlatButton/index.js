/**
 *
 * StyledFlatButton
 *
 */
import styled from 'styled-components';
import FlatButton from 'material-ui/FlatButton';


const StyledFlatButton = styled(FlatButton)`
  color: #366 !important;

  & svg {
    fill: #d2d2c3 !important;
    transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
  }

  &:hover svg {
    fill: #366 !important;
  }
`;

StyledFlatButton.propTypes = {};

export default StyledFlatButton;
