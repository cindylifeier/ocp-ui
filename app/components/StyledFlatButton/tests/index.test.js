import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import 'jest-styled-components';
import { MuiThemeProvider } from 'material-ui';
import ContentAddCircle from '@material-ui/icons/AddCircle';

import StyledFlatButton from '../index';

configure({ adapter: new Adapter() });

describe('<StyledFlatButton />', () => {
  describe('snapshot tests', () => {
    it('should match snapshot', () => {
      // Arrange
      const label = 'label';

      // Act
      const renderedComponent = shallow(<StyledFlatButton label={label} icon={<ContentAddCircle />} />);

      // Assert
      expect(renderedComponent).toMatchSnapshot();
    });
  });

  describe('style tests', () => {
    it('should have color style', () => {
      // Arrange
      const label = 'label';

      // Act
      const renderedComponent = mount(<MuiThemeProvider><StyledFlatButton label={label} /></MuiThemeProvider>);

      // Assert
      expect(renderedComponent.find(StyledFlatButton)).toHaveStyleRule('color', '#366 !important');
    });

    it('should have svg styles', () => {
      // Arrange
      const label = 'label';

      // Act
      const renderedComponent = mount(<MuiThemeProvider><StyledFlatButton label={label} /></MuiThemeProvider>);

      // Assert
      expect(renderedComponent.find(StyledFlatButton)).toHaveStyleRule('fill', '#d2d2c3 !important', {
        modifier: ' svg',
      });
      expect(renderedComponent.find(StyledFlatButton)).toHaveStyleRule('transition', 'all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms', {
        modifier: ' svg',
      });
    });

    it('should have svg styles in hover state', () => {
      // Arrange
      const label = 'label';

      // Act
      const renderedComponent = mount(<MuiThemeProvider><StyledFlatButton label={label} /></MuiThemeProvider>);

      // Assert
      expect(renderedComponent.find(StyledFlatButton)).toHaveStyleRule('fill', '#366 !important', {
        modifier: ':hover svg',
      });
    });
  });
});
