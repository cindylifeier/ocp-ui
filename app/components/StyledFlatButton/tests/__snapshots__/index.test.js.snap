// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledFlatButton /> snapshot tests should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StyledFlatButton__StyledFlatButton
    icon={<pure(AddCircle) />}
    label="label"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "className": "StyledFlatButton__StyledFlatButton-jENDUq hlOZwo",
      "disabled": false,
      "fullWidth": false,
      "icon": <pure(AddCircle) />,
      "label": "label",
      "labelPosition": "after",
      "labelStyle": Object {},
      "onKeyboardFocus": [Function],
      "onMouseEnter": [Function],
      "onMouseLeave": [Function],
      "onTouchStart": [Function],
      "primary": false,
      "secondary": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "StyledFlatButton__StyledFlatButton-jENDUq hlOZwo",
        "disabled": false,
        "fullWidth": false,
        "icon": <pure(AddCircle) />,
        "label": "label",
        "labelPosition": "after",
        "labelStyle": Object {},
        "onKeyboardFocus": [Function],
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
        "onTouchStart": [Function],
        "primary": false,
        "secondary": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
